lib/charms/layer/basic.py

```diff
@@ -278,6 +284,29 @@ def _load_wheelhouse_versions():
     return versions
 
 
+def _add_back_versions(pkgs, _versions):
+    """Add back the version strings to each of the packages.
+
+    The versions are LooseVersion() from _load_wheelhouse_versions().  This
+    function strips the ".zip" or ".tar.gz" from the end of the version string
+    and adds it back to the package in the form of <package_name>==<version>
+    """
+    def _strip_ext(s):
+        _s = str(s)
+        for ending in [".zip", ".tar.gz"]:
+            if _s.endswith(ending):
+                return _s[:-len(ending)]
+        return _s
+
+    def _maybe_make_version(k):
```

Comment:
This is a bit hard to read on the overall, because of the one-char var names and the missing docstrings+types. I know docstrings aren't mandatory here but they would really help me and my weak brain.

So I think `k` is a package name that may or may not have an extension and may or may not have a version number, correct?

Reply:
Yes, sometimes a function can be a bit opaque.  The function would be better named as `_maybe_add_version_to_pkg` as that is what it is doing (i.e. if the pkg has a version in the versions{} then it will add it, otherwise it will leave it alone.